lib/json.rs

```diff
@@ -45,32 +45,46 @@ pub fn to_json_value<T: serde::Serialize>(x: T) -> ddlog_std::Result<JsonValue,
     res2std(serde_json::to_value(x).map(JsonValue::from))
 }
 
-pub struct ValueWrapper(serde_json::value::Value);
-
-impl serde::Serialize for ValueWrapper {
+impl serde::Serialize for JsonValue {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: serde::Serializer,
     {
         if serializer.is_human_readable() {
```

Comment:
Do we need to explicitly handle this? Doesn't serde's infrastructure do the oh its own?

Reply:
This code could use some documentation, but it's there for a reason. The problem is that `serde::json::Value` cannot be deserialized from non-self-describing formats like bincode.  As a workaround, we first serialize the JSON value into a JSON string and store the string in bincode.  The next problem for which there is currently no good solution is that there is no way for the `Serialize`/`Deserialize` implementation to tell whether it is working with a self-describing format.  We use `is_human_readable` as a very rough and generally incorrect approximation of that.